File: PracticasProgramacion/Practica_13/cpp/Persona.cpp
#include "Persona.h"

Persona::Persona() : nombre_(""), ap_(""), am_(""), genero_(""), edad_(0) {}

Persona::Persona(const std::string& nombre, const std::string& ap, const std::string& am,
                 const std::string& genero, int edad)
    : nombre_(nombre), ap_(ap), am_(am), genero_(genero), edad_(edad) {}

std::string Persona::getNombre() const { return nombre_; }
void Persona::setNombre(const std::string& nombre) { nombre_ = nombre; }
std::string Persona::getAp() const { return ap_; }
void Persona::setAp(const std::string& ap) { ap_ = ap; }
std::string Persona::getAm() const { return am_; }
void Persona::setAm(const std::string& am) { am_ = am; }
std::string Persona::getGenero() const { return genero_; }
void Persona::setGenero(const std::string& genero) { genero_ = genero; }
int Persona::getEdad() const { return edad_; }
void Persona::setEdad(int edad) { edad_ = edad; }

File: PracticasProgramacion/Practica_07/python/persona.py
"""Clase Persona: nombre, ap, am, genero, edad."""


class Persona:
    def __init__(
        self,
        nombre: str = "",
        ap: str = "",
        am: str = "",
        genero: str = "",
        edad: int = 0,
    ) -> None:
        self._nombre = nombre
        self._ap = ap
        self._am = am
        self._genero = genero
        self._edad = edad

    def get_nombre(self) -> str:
        return self._nombre

    def set_nombre(self, nombre: str) -> None:
        self._nombre = nombre

    def get_ap(self) -> str:
        return self._ap

    def set_ap(self, ap: str) -> None:
        self._ap = ap

    def get_am(self) -> str:
        return self._am

    def set_am(self, am: str) -> None:
        self._am = am

    def get_genero(self) -> str:
        return self._genero

    def set_genero(self, genero: str) -> None:
        self._genero = genero

    def get_edad(self) -> int:
        return self._edad

    def set_edad(self, edad: int) -> None:
        self._edad = edad
